core/io/marshalls.cpp

```diff
@@ -681,8 +681,6 @@ Error decode_variant(Variant &r_variant, const uint8_t *p_buffer, int p_len, int
 
 				if (r_len)
 					(*r_len) += adv;
-				len -= adv;
-				buf += adv;
```

Comment:
I think these were added to make sure that the `len` and `buf` have the right values if they were to be reused in newly added code in this code block. It's more important to have safe values for buffer lengths, even if the assignment is pointless in the *current* code, that to risk out of bound reads in the future.

I think we already discussed this change in a similar PR a while ago :)

If we really want to silence this static analyzer warning, I guess this code could be commented out with an explanatory comment saying to uncomment if new code using those is added.

Reply:
I was maybe thinking of https://github.com/godotengine/godot/pull/19963#discussion_r200141487

Reply:
I was pretty worried by these changes too, but after a closer look I noticed this "safe" behaviour is not respected consistently in the file, so dropping it completely might be a better choice after all.

In general, not updating `len` and `buf` (together) will result in reading from an earlier starting point, so it's unlikely it could be coded in a way that goes outside of boundaries (unless one mistakenly update `buf` but not `len`, which could happen anyway). So, I would approve this change for marshalls.